### PR TITLE
Ron/tsk 3326 sumo logic integration

### PR DESCRIPTION
**Pull request by Mono2202**

### CODE REVIEW

File: gen/content/metadata.yaml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# This is a generated file. DO NOT EDIT.
+name: ""
```
Comment (ozhey) [inline gen/content/metadata.yaml:2]: What's that? I think it can be removed

File: plugin.yaml [hunk 2]
```diff
@@ -66,13 +66,15 @@ connection_types:
   prometheus:
     reference: prometheus
   rapid7:
-    refernce: rapid7
+    reference: rapid7
```
Comment (ozhey) [inline plugin.yaml:69]: good catch

File: plugins/sumo-logic/mask.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+connection_type:
+  name: sumo-logic
+  oauth_supported: false
+  attributes:
+    REQUEST_URL:
```
Comment (ozhey) [inline plugins/sumo-logic/mask.yaml:5]: This is going to have to change (probably tomorrow) to `api_address` due to a migration we are doing (to make everything use the same named convention, lowercase & snakecase)
@IdoGafi is working on it, sync with him

File: plugins/sumo-logic/sumo-logic.go [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+)
+
+type SumoLogicPlugin struct{}
+
+func (p SumoLogicPlugin) HandleAuth(req *http.Request, conn map[string]string) error {
+	return connections.HandleGenericConnection(conn, req, nil, nil)
+}
+
+func (p SumoLogicPlugin) TestConnection(connection *blink_conn.ConnectionInstance) (bool, []byte) {
+	requestUrl, ok := connection.Data["REQUEST_URL"]
```
Comment (ozhey) [inline plugins/sumo-logic/sumo-logic.go:17]: same comment here about `api_address` & also use the constant from `consts` package

**Review by ozhey — COMMENTED** [review]
Great job!
I recommend adding .idea to your global .gitignore file (if you don't have one, add one :-) ), we don't want to commit such files
And of course we'll talk tomorrow about which actions to keep

**Review by blink-daniel — CHANGES_REQUESTED** [review]
### Great job there, impressive how fast you got up to speed :)

Few more comment-

1. your generated content folder contain all actions from the open api file. looking into your mask file, I see that you only want to add one action, so I guess the other actions are from a previous attempt to call the make named-actions cmd? if that's the case, delete the gen folder and re-run the make so only the one action you wanted will be added. also, the gen/content/metadata.yaml file was probably generated from a failed attempt earlier due to padding issues in your mask file, so re-running the make should solve this as well. in the end, make sure that 0003_actions folder contain only the actions that we want to expose to the user from the sumo logic api.
2. Please use your real name as your git author name - think of someone else seeing that "Mono" committed a change wondering "who is this guy ?" :)

**Review by blink-daniel — APPROVED** [review]

**Comment by blink-daniel** [comment]
looks great, after you verify we have all actions we want + test this in an deployed env. we can go ahead and push to master.
regarding our merge strategy - we use a squash merge strategy in our team so the history will be cleaner + cherry picking features / bisecting them will be easier.

File: plugins/sumo-logic/mask.yaml [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+      required: true
+collection:
+  name: sumo-logic
+  icon_uri: "https://app.dev.blinkops.com/assets/images/Icons/Companies/Company/XL/SumoLogic.svg"
+  active: true
+is_connection_optional: false
+request_url_conn_type: sumo-logic
+actions:
+  createUser:
+    alias: CreateUser
```
Comment (ozhey) [inline plugins/sumo-logic/mask.yaml:37]: No need for aliases, you can leave this one empty. If the autogenerated display name isn't clear, you can change it instead (`display_name` and not `alias`)

File: plugins/sumo-logic/mask.yaml [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+    parameters:
+      firstName:
+        alias: First Name
```
Comment (ozhey) [inline plugins/sumo-logic/mask.yaml:40]: This is redundant, display names are generated automatically by putting spaces between words and camelcasing everything. Only add `alias` if you want it to be really different

File: plugins/sumo-logic/mask.yaml [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+        type: string
```
Comment (ozhey) [inline plugins/sumo-logic/mask.yaml:41]: If the type is the same in the openapi definition, this is redundant. Only include it when you want to override the type defined in the openapi

File: plugins/sumo-logic/mask.yaml [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+        required: true
```
Comment (ozhey) [inline plugins/sumo-logic/mask.yaml:42]: same here, if it's already required in the openapi, you don't need to specifiy it again.

**Pull request merged**